pyomo/core/tests/unit/test_indexed_slice.py

```diff
@@ -548,5 +548,48 @@ def test_duplicate(self):
         self.assertEqual(a._len+1, b1._len)
         self.assertEqual(hash(b), hash(b1))
 
+    def test_invalid_slices(self):
+        m = self.m
+        m.x = Var()
+        for var in m.x[:]:
+            self.assertIs(var, m.x)
+
+        with self.assertRaisesRegexp(
+            IndexError, 'Index .* contains an invalid number of '
+            'entries for component .*'):
+            _slicer = m.b[:]
+
+        with self.assertRaisesRegexp(
+            IndexError, 'Index .* contains an invalid number of '
+            'entries for component .*'):
+            _slicer = m.b[:, :, :]
+
+        with self.assertRaisesRegexp(
+            IndexError, 'Index .* contains an invalid number of '
+            'entries for component .*'):
+            _slicer = m.b[:].c[...].x
+
+        with self.assertRaisesRegexp(
+            IndexError, 'Index .* contains an invalid number of '
+            'entries for component .*'):
+            _slicer = m.b[:].c[:,:,:].x
+
+        with self.assertRaisesRegexp(
+            IndexError, 'Index .* contains an invalid number of '
+            'entries for component .*'):
+            _slicer = m.b[2, :].c[:,:,:].x
+            # Error not raised immediately because accessing c is deferred
+            # until iteration.
+            for var in _slicer:
+                pass
```

Comment:
This is fine, but if you want a single-line alternative, `list(_slicer)` ought to work.

Reply:
Thanks, update.